### DISCUSSION TITLE
Assets - 'Allow uploading directly to the field' unexpected results

### DISCUSSION THREAD

**Discussion by anchovy** [craftcms/cms#10145] [Ideas] [answered]
@brandonkelly Continuing on from #10139 

Ok, if this is the expected behaviour then that's fine - I've found with editors for some reason, who have full access, keep using this function to upload the file and we've ended up with literally hundreds of the same file, different names, used across the site - a nightmare to manage.

So - what about having a config setting where we can set this option to unchecked by default?

Add a setting (https://craftcms.com/docs/3.x/config/config-settings.html#assets) i.e. allowUploadingDirectlyToField

```
allowUploadingDirectlyToField
Allowed types: boolean(opens new window)
Default value: true
Defined by: GeneralConfig::$allowUploadingDirectlyToField
```

This way - the default behaviour doesn't change, but it allows us to override the option being checked by default?

**Comment by brandonkelly** [comment, marked as answer]
You can disable the “Allow uploading directly to the field” setting for all existing Assets fields by doing the following:

1. Run the command:

   ```sh
   php craft project-config/rebuild
   ```

2. In a text editor, use the Find & Replace feature to replace all instances of `allowUploads: true` with `allowUploads: false` in your `config/project/` folder.

3. Run the following commands:

   ```sh
   php craft project-config/touch
   ```

   ```sh
   php craft project-config/apply
   ```

Adding a new config setting would probably just make things more confusing. In general we are trying to move away from config settings that conflict with other settings.

**Reply by anchovy** [reply]
Thanks for the reply @brandonkelly - how would this this new config setting conflict with another setting, which setting? 
It would simply allow dev's to change the default checked status 'Allow uploading directly to the field' option. If they do change the default _true_ to _false_ in the config they could still enable this for a particular asset field as required.

I also think we need the warning about an existing file and the option to use the existing file or rename the new file.
At least this way, the user know's there is an existing file and they can then review this file to see if it's correct or needs to be updated, or if they do need to upload a totally different file and change the file name to be more descriptive.

**Reply by brandonkelly** [reply]
> which setting?

Assets fields’ “Allow uploading directly to the field” settings.

> I also think we need the warning about an existing file and the option to use the existing file or rename the new file.

Yeah that’s the suggestion I was thinking you should create a discussion for. Can you create a separate post for it?

**Reply by anchovy** [reply]
> which setting?
>> Assets fields’ “Allow uploading directly to the field” settings.

Sorry - I was meaning which setting, if this was added, would it conflict with?
> Adding a new config setting would probably just make things more confusing. In general we are trying to move away from config settings that conflict with other settings.

> Yeah that’s the suggestion I was thinking you should create a discussion for. Can you create a separate post for it?

Will do.
